File: mRemoteV1/Root/PuttySessions.Info.vb

Imports mRemote3G.Tools

Namespace Root

    Namespace PuttySessions
        Public Class PuttySessionsInfo
            Inherits Info

            Public Sub New()
                MyBase.New(RootType.PuttySessions)
            End Sub

            Private _name As String = Language.Language.strPuttySavedSessionsRootName

            <LocalizedAttributes.LocalizedDefaultValue("strPuttySavedSessionsRootName")>
            Public Overrides Property Name As String
                Get
                    Return _name
                End Get
                Set
                    If _name = value Then Return
                    _name = value
                    If TreeNode IsNot Nothing Then
                        TreeNode.Text = value
                    End If
                    My.Settings.PuttySavedSessionsName = value
                End Set
            End Property

            Private _panel As String = Language.Language.strGeneral

            <LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1),
                LocalizedAttributes.LocalizedDisplayName("strPropertyNamePanel"),
                LocalizedAttributes.LocalizedDescription("strPropertyDescriptionPanel")>
            Public Property Panel As String
                Get
                    Return _panel
                End Get
                Set
                    If _panel = value Then Return
                    _panel = value
                    My.Settings.PuttySavedSessionsPanel = value
                End Set
            End Property
        End Class
    End Namespace

End Namespace

File: mRemoteV1/CS/Root/PuttySessions.Info.cs
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
using mRemoteNG.My;
using mRemoteNG.Tools;


namespace mRemoteNG.Root.PuttySessions
{
	public class Info : Root.Info
	{
		public Info() : base(RootType.PuttySessions)
		{
			// VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
			_name = Language.strPuttySavedSessionsRootName;
			_panel = My.Language.strGeneral;
		}
				
		private string _name; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
        [LocalizedAttributes.LocalizedDefaultValue("strPuttySavedSessionsRootName")]
        public override string Name
		{
			get
			{
				return _name;
			}
			set
			{
				if (_name == value)
				{
					return ;
				}
				_name = value;
				if (TreeNode != null)
				{
					TreeNode.Text = value;
				}
                My.Settings.Default.PuttySavedSessionsName = value;
			}
		}
				
		private string _panel; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
        [LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1),
            LocalizedAttributes.LocalizedDisplayName("strPropertyNamePanel"),
            LocalizedAttributes.LocalizedDescription("strPropertyDescriptionPanel")]
        public string Panel
		{
			get
			{
				return _panel;
			}
			set
			{
				if (_panel == value)
				{
					return ;
				}
				_panel = value;
				My.Settings.Default.PuttySavedSessionsPanel = value;
			}
		}
	}
}